tests/sync/app.cpp

```diff
@@ -824,6 +827,368 @@ TEST_CASE("app: link_user integration", "[sync][app]") {
+        });
+        
+        CHECK(processed);
+    }
+    
+    SECTION("find") {
+        bool processed = false;
+        
+        collection.find(dog_document,
+                        [&](std::string document_json, Optional<app::AppError> error) {
+            CHECK(!error);
+            auto json = nlohmann::json::parse(document_json);
+            CHECK(json.is_array());
+        });
+        
+        collection.find_one(dog_document,
+                            [&](Optional<std::string> document_json, Optional<app::AppError> error) {
+            CHECK(!error);
+            CHECK(!document_json);
+        });
+        
+        collection.insert_one(dog_document,
+                              [&](std::string, Optional<app::AppError> error) {
+            CHECK(!error);
+        });
+        
+        collection.find(dog_document,
+                        [&](std::string document_json, Optional<app::AppError> error) {
+            CHECK(!error);
+            auto json = nlohmann::json::parse(document_json);
+            CHECK(json.is_array());
+        });
+        
+        realm::app::RemoteMongoCollection::RemoteFindOptions options {
+            1, //document limit
+            util::Optional<std::string>(nlohmann::json({{"name", "fido"}}).dump()), //project
+            util::Optional<std::string>(nlohmann::json({{"name", -1}}).dump()) //sort
+        };
+        
+        collection.find(dog_document, options, [&](std::string document_json, Optional<app::AppError> error) {
+            CHECK(!error);
+            auto json = nlohmann::json::parse(document_json);
+            CHECK(json.is_array());
+            auto json_array = json.get<std::vector<nlohmann::json>>();
+            CHECK(json_array.size() == 1);
+        });
+        
+        collection.find_one(dog_document,
+                            [&](Optional<std::string> document_json, Optional<app::AppError> error) {
+            CHECK(!error);
+            auto json = nlohmann::json::parse(*document_json);
+            auto name = json.at("name").get<std::string>();
+            CHECK(name == "fido");
+        });
+        
+        collection.find_one(dog_document, options, [&](Optional<std::string> document_json, Optional<app::AppError> error) {
+            CHECK(!error);
+            auto json = nlohmann::json::parse(*document_json);
+            auto name = json.at("name").get<std::string>();
+            CHECK(name == "fido");
+        });
+        
+        realm::app::RemoteMongoCollection::RemoteFindOneAndModifyOptions find_and_modify_options {
+            util::Optional<std::string>(nlohmann::json({{"name", "fido"}}).dump()), //project
+            util::Optional<std::string>(nlohmann::json({{"name", 1}}).dump()), //sort,
+            true, //upsert
+            true // return new doc
+        };
+        
+        collection.find_one_and_delete(dog_document, find_and_modify_options, [&](Optional<app::AppError> error) {
+            CHECK(!error);
```

Comment:
check that the result is valid and has been updated

Reply:
I'm now doing a check to see if the size of the collection is 0 after the delete